File: 27.remove-element.ts
/*
 * @lc app=leetcode id=27 lang=typescript
 *
 * [27] Remove Element
 */

// @lc code=start
function removeElement(nums: any[], val: number): number {
  let dups = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] === val) {
      for (let j = nums.length - 1; j > i; j--) {
        if (nums[j] !== val) {
          let temp = nums[j];
          nums[j] = nums[i];
          nums[i] = temp;
          break;
        }
      }
    }
  }
  for (let k = 0; k < nums.length; k++) {
    if (nums[k] === val) dups++;
  }
  return nums.length - dups;
}
// @lc code=end;

File: 27.remove-element.py
#
# @lc app=leetcode id=27 lang=python3
#
# [27] Remove Element
#

# @lc code=start
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        L = 0
        for R in range(len(nums)):
            if nums[R] != val:
                nums[L] = nums[R]
                L += 1
        return L
# @lc code=end
